compiler/qsc_doc_gen/src/generate_docs/tests.rs

```diff
@@ -211,3 +278,115 @@ fn top_index_file_generation() {
     "#]]
     .assert_eq(full_contents.as_str());
 }
+
+#[test]
+fn dependency_with_main_namespace_fully_qualified_name() {
+    let full_contents = check_doc_generation(
+        Some(("dep/Main.qs", "namespace Main { operation DependencyFunction() : Unit {} export DependencyFunction; }")),
+        Some("MyDep"),
+        ("src/Main.qs", "operation Main() : Unit { MyDep.DependencyFunction() }"),
+        "DependencyFunction.md",
+    );
+
+    // The fully qualified name should be "MyDep.DependencyFunction", not "MyDep.Main.DependencyFunction"
+    // After the fix, "Main" namespace should be omitted for aliased packages
+    expect![[r#"
+        ---
+        uid: Qdk.Main.DependencyFunction
+        title: DependencyFunction operation
+        description: "Q# DependencyFunction operation: "
+        ms.date: {TIMESTAMP}
+        qsharp.kind: operation
+        qsharp.package: MyDep
+        qsharp.namespace: Main
+        qsharp.name: DependencyFunction
+        qsharp.summary: ""
+        ---
+
+        # DependencyFunction operation
+
+        Fully qualified name: MyDep.DependencyFunction
+
+        ```qsharp
+        operation DependencyFunction() : Unit
+        ```
+    "#]]
+    .assert_eq(full_contents.as_str());
+}
+
+#[test]
+fn dependency_with_non_main_namespace_fully_qualified_name() {
+    let full_contents = check_doc_generation(
+        Some((
+            "dep/Utils.qs",
+            "namespace Utils { operation UtilityFunction() : Unit {} export UtilityFunction; }",
+        )),
+        Some("MyDep"),
+        (
+            "src/Main.qs",
+            "operation Main() : Unit { MyDep.Utils.UtilityFunction() }",
+        ),
+        "UtilityFunction.md",
+    );
+
+    // For non-Main namespaces, the namespace should be included in fully qualified name
+    expect![[r#"
+        ---
+        uid: Qdk.Utils.UtilityFunction
+        title: UtilityFunction operation
+        description: "Q# UtilityFunction operation: "
+        ms.date: {TIMESTAMP}
+        qsharp.kind: operation
+        qsharp.package: MyDep
+        qsharp.namespace: Utils
+        qsharp.name: UtilityFunction
+        qsharp.summary: ""
+        ---
+
+        # UtilityFunction operation
+
+        Fully qualified name: MyDep.Utils.UtilityFunction
+
+        ```qsharp
+        operation UtilityFunction() : Unit
+        ```
+    "#]]
+    .assert_eq(full_contents.as_str());
+}
+
+#[test]
+fn user_code_with_main_namespace_fully_qualified_name() {
+    let full_contents = check_doc_generation(
+        None, // No dependencies
+        None,
+        (
+            "src/Main.qs",
+            "namespace Main { operation UserFunction() : Unit {} export UserFunction; }",
+        ),
+        "UserFunction.md",
+    );
+
+    // For user code (PackageKind::UserCode), Main namespace should still be included
```

Comment:
Just want to double-check that this is the behavior we want. Seems to me like we shouldn't have the `Main` namespace even in user-code, but I could be convince otherwise.

Reply:
You do need the `Main` namespace when calling user code from user code, so I thought it was good to leave it in. I agree that it's a bit ambiguous what you _want_ - if your user code was a library, you were using this function to generate documentation for consumers, you may want to do something different with `Main`. 

But currently this is only wired up to the "Show API Documentation" command in VS Code. In that case, "user code" is the current project, so if you want to call something from `Main` you _do_ need the `Main` namespace.

Ultimately I think this is just a bit of a gap in the package ecosystem design altogether... and this is the best we can do in doc generation.